src/mapping/cellranger_multi/script.py

```diff
@@ -116,16 +112,59 @@ def setup_logger():
                    if cellranger_param == "reference")
 LIBRARY_PARAMS = tuple(LIBRARY_CONFIG_KEYS.keys())
 SAMPLE_PARAMS = tuple(SAMPLE_PARAMS_CONFIG_KEYS.keys())
+HELPER_INPUT = {
+    'gex_input': 'Gene Expression',
+    'abc_input': 'Antibody Capture',
+    'cgc_input': 'CRISPR Guide Capture',
+    'mux_input': 'Multiplexing Capture',
+    'vdj_input': 'VDJ',
+    'vdj_t_input': 'VDJ-T',
+    'vdj_t_gd_input': 'VDJ-T-GD',
+    'vdj_b_input': 'VDJ-B',
+    'agc_input': 'Antigen Capture'
+}
 
 
+def infer_library_id_from_path(input_path: str) -> str:
+    match = re.match(fastq_regex, input_path)
+    assert match is not None, \
+        f"File name of '{input_path}' should match regex {fastq_regex}."
+    return match.group(1)
+
+def transform_helper_inputs(par: dict[str, any]) -> dict[str, any]:
+    helper_input = {
+        "input": [],
+        "library_id": [],
+        "library_type": []
+    }
+    for input_type, library_type in HELPER_INPUT.items():
+        if par[input_type]:
+            paths = [Path(fastq) for fastq in par[input_type]]
```

Comment:
Can you move this code:

```python

        par["input"] = [Path(fastq) for fastq in par["input"]]
        if len(par["input"]) == 1 and par["input"][0].is_dir():
            logger.info("Detected '--input' as a directory, "
                        "traversing to see if we can detect any FASTQ files.")
            par["input"] = [input_path for input_path in par["input"][0].rglob('*')
                            if re.match(fastq_regex, input_path.name) ]

        # check input fastq files
        for input_path in par["input"]:
            assert re.match(fastq_regex, input_path.name) is not None, \
                f"File name of --input '{input_path}' should match regex {fastq_regex}."
```

to a helper function and use it in `transform_helper_inputs` and `process_params` ?